app/src/main/java/org/mozilla/tv/firefox/ext/Js.kt

```diff
@@ -90,6 +91,18 @@ var _firefoxTV_isPlaybackStateObserverLoaded;
         javaInterface.syncPlaybackState(isVideoPresent, isPlaying, positionSeconds, playbackRate);
     }
 
+    function adjustVideoCSS() {
+        const rawVideoElements = document.getElementsByTagName('video');
+        // Extract elements from HTMLCollection and collect them into an Array
+        const videoElements = new Array(...rawVideoElements);
+
+        videoElements.forEach((vid) => {
+            vid.style.left = '0px';
+            vid.style.transform = 'translate(0px, 0px)';
```

Comment:
It seems risky to modify the CSS unconditionally in the event that web devs are doing something less innocuous using these CSS parameters. My thinking was that we'd only do this if we saw the conditions that seem to cause problems: that is `left 50% + translateX -50%` (or maybe something slightly broader).

What do you think? Does that seem reasonable?

Reply:
Seems reasonable to me.

Reply:
This turned out to be a lot more frustrating than expected.  Finding `left` involves dividing `computedLeft` by the `width` of the parent element, which is annoying but fairly trivial.  Checking transform is substantially more complex (querying `transform` on one of these broken sites returns the string `"matrix(1, 0, 0, 1, -362, 0)"`, and I'm not even yet sure if that pattern holds true for all examples).

I spent a full day QAing this change, and it didn't seem to negatively impact any sites.  We need a rendering engine fix for the long term, but for the short term I don't think it's a good use of time to keep building up CSS injection logic.  This code is necessary for us right now, but it's just inherently really, really bad.  I think we should ship this code, or cut the feature and wait on the rendering engine fix.

Reply:
via https://stackoverflow.com/a/42267468, it looks like we can get the translateX pixel value (through the transform matrix) by:
```js
var style = window.getComputedStyle(myElement);
var matrix = new DOMMatrix(style.transform);
console.log('translateX: ', matrix.m41);
```

afaict (disclaimer: I'm slightly distracted), we can identify if it's "translateX(-50%)" by comparing this translateX value against `style.transformOrigin` (where we're translating from) which looks like, "362px 203.633px" (this value might be equivalent to left, I'm not sure). If transformOrigin X is equivalent to that previous translate X value, we're likely in a `translateX(-50%)` situation. I think that's everything we need.

> I spent a full day QAing this change, and it didn't seem to negatively impact any sites.

Without conditionally making this change, we have a high probability of breaking page layout on some site. I'm concerned that the sites that break will affect our non-YT users and they'll leave. Given that our users have a low frequency of reporting bugs and our limited ability to test a wide range of sites, I'm also concerned that we won't find these bugs for a while, encouraging more users to leave.

> but for the short term I don't think it's a good use of time to keep building up CSS injection logic.

I don't think it's a much larger investment in time to add the conditional statement. I agree with your sentiment – that CSS injection is fragile, risky, and generally not a good idea – but I think the conditional change actually makes it less risky and a better change because it'll affect fewer pages and we can still confirm it fixes the specific sites we expect it to.

Let's follow-up IRL. I think we should pair through building the conditional statement to increase our confidence in the solution.

Reply:
You're right, and that's a better approach.  Thank you for suggesting it, we can work through it later today.